AppCenter/AppCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -82,8 +103,16 @@ - (int)executeQueryUsingBlock:(MSDBStorageQueryBlock)callback {
   // The value is stored as part of the database connection and must be reset every time the database is opened.
   long maxPageCount = self.maxSizeInBytes / self.pageSize;
   result = [MSDBStorage setMaxPageCount:maxPageCount inOpenedDatabase:db];
+
+  // Do not proceed with the query if the database is corrupted.
+  if (result == SQLITE_CORRUPT || result == SQLITE_NOTADB) {
+    sqlite3_close(db);
+    return result;
+  }
+
+  // Log a warning if max page count can't be set.
   if (result != SQLITE_OK) {
-    MSLogError([MSAppCenter logTag], @"Failed to open database with specified maximum size constraint.");
+    MSLogWarning([MSAppCenter logTag], @"Failed to open database with specified maximum size constraint.");
```

Comment:
Why was this downgraded to a warning?  Also I think we should include the result code in the log here

Reply:
Or if there are logs where the method is used I suppose we could put the result there.

Reply:
The code is already printed. "warning" because in this case, the main query will be executed

Reply:
I think it should stay as an error level. If the operation fails, that means it won't use a limit that customers set.